src/libraries/Common/tests/System/Net/Prerequisites/RemoteLoopServer/Handlers/RemoteLoopHandler.cs

```diff
@@ -0,0 +1,155 @@
+
+                // now we are connected, pump messages
+                bool close = false;
+                while (!close)
+                {
+                    // wait for either message
+                    await Task.WhenAny(testedNext, controlNext).ConfigureAwait(false);
+
+                    if (testedNext.IsCompleted)
+                    {
+                        if (testedNext.Result > 0)
+                        {
+                            var slice = new ArraySegment<byte>(testedBuffer, 0, testedNext.Result);
+                            await control.SendAsync(slice, WebSocketMessageType.Binary, true, cts.Token).ConfigureAwait(false);
+                        }
+                        if (testedNext.IsCanceled || testedNext.IsFaulted || !tested.Connected)
+                        {
+                            close = true;
+                        }
+                        else
+                        {
+                            testedNext = tested.ReceiveAsync(new Memory<byte>(testedBuffer), SocketFlags.None, cts.Token).AsTask();
+                        }
+                    }
+                    if (controlNext.IsCompleted)
+                    {
+                        if (controlNext.Result.Count > 0)
+                        {
+                            var slice = new ArraySegment<byte>(controlBuffer, 0, controlNext.Result.Count);
+                            await tested.SendAsync(slice, SocketFlags.None, cts.Token).ConfigureAwait(false);
+                        }
+                        if (controlNext.IsCanceled || controlNext.IsFaulted || controlNext.Result.MessageType == WebSocketMessageType.Close)
+                        {
+                            close = true;
+                        }
+                        else
+                        {
+                            controlNext = control.ReceiveAsync(new ArraySegment<byte>(controlBuffer), cts.Token);
+                        }
+                    }
+                }
+                if (tested.Connected)
+                {
+                    tested.Disconnect(false);
+                }
+                if (control.State == WebSocketState.Open || control.State == WebSocketState.Connecting || control.State == WebSocketState.None)
+                {
+                    try
+                    {
+                        await control.CloseAsync(WebSocketCloseStatus.NormalClosure, "closing remoteLoop", cts.Token).ConfigureAwait(false);
+                    }
+                    catch (WebSocketException)
+                    {
+                        // We might want to log these exceptions. But for now we ignore them.
+                    }
+                }
+                cts.Cancel();
+            }
+            catch (Exception)
+            {
+                // We might want to log these exceptions. But for now we ignore them.
```

Comment:
Logging these could be helpful when debugging issues with connections.

Reply:
okay